c_src/sqlite3_nif.c

```diff
@@ -12,6 +12,36 @@
 
 #define MAX_ATOM_LENGTH 255
 
+static ERL_NIF_TERM am_ok;
+static ERL_NIF_TERM am_error;
+static ERL_NIF_TERM am_nil;
+static ERL_NIF_TERM am_out_of_memory;
+static ERL_NIF_TERM am_done;
+static ERL_NIF_TERM am_row;
+static ERL_NIF_TERM am_rows;
+static ERL_NIF_TERM am_invalid_filename;
+static ERL_NIF_TERM am_invalid_flags;
+static ERL_NIF_TERM am_database_open_failed;
```

Comment:
This should be rc + exception, since otherwise it's hard to understand why the open failed. Could be user permissions, could be something else. I think I actually had a problem with this one in the past.

Reply:
Oh I saw this being done in the erlang sqlite implementation and didn't understand the optimization being done when I initially implemented this. So I just created the atoms on demand rather than all up front at once.